pkg/karmadactl/rebalance/rebalance.go

```diff
@@ -0,0 +1,251 @@
+			if err := opts.Complete(f, args); err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Run(f, args); err != nil {
+				return err
+			}
+			return nil
+		},
+		Annotations: map[string]string{
+			util.TagCommandGroup: util.GroupAdvancedCommands,
+		},
+	}
+
+	flag := cmd.Flags()
+	opts.AddFlags(flag)
+	options.AddKubeConfigFlags(flag)
+
+	return cmd
+}
+
+// CommandRebalanceOption holds all command options for rebalance
+type CommandRebalanceOption struct {
+	// DryRun tells if run the command in dry-run mode, without making any server requests.
+	DryRun bool
+
+	// Namespace is the namespace of resource
+	Namespace string
+
+	resource.FilenameOptions
+}
+
+// AddFlags adds flags to the specified FlagSet.
+func (o *CommandRebalanceOption) AddFlags(flags *pflag.FlagSet) {
+	flags.BoolVar(&o.DryRun, "dry-run", false, "Run the command in dry-run mode, without making any server requests.")
+	flags.StringVarP(&o.Namespace, "namespace", "n", o.Namespace, "If present, the namespace scope for this CLI request")
+}
+
+// Complete ensures that options are valid and marshals them if necessary
+func (o *CommandRebalanceOption) Complete(f util.Factory, args []string) error {
+	if len(args) != 2 {
+		return fmt.Errorf("incorrect command format, please use correct command format")
+	}
+
+	if o.Namespace == "" {
+		var err error
+		o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+		if err != nil {
+			return fmt.Errorf("failed to get namespace from Factory. error: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Validate checks to the RebalanceOptions to see if there is sufficient information run the command
+func (o *CommandRebalanceOption) Validate() error {
+	return nil
+}
+
+// Run rebalance resources to achieve a fresh rescheduling
+func (o *CommandRebalanceOption) Run(f util.Factory, args []string) error {
+	// 1. get target rebalance object
+	obj, err := o.getObjInfo(f, args)
+	if err != nil {
+		return fmt.Errorf("failed to get resource in Karmada: %w", err)
+	}
+
+	// 2. get karmada-apiserver client
+	karmadaClient, err := f.KarmadaClientSet()
+	if err != nil {
+		return fmt.Errorf("failed to get karmada-apiserver client. err: %w", err)
+	}
+
+	// 3. rebalance object
+	return o.rebalanceObject(karmadaClient, obj)
+}
+
+// getObjInfo get obj info in member cluster
+func (o *CommandRebalanceOption) getObjInfo(f cmdutil.Factory, args []string) (*unstructured.Unstructured, error) {
+	r := f.NewBuilder().
+		Unstructured().
+		NamespaceParam(o.Namespace).
+		FilenameParam(false, &o.FilenameOptions).
+		RequestChunksOf(500).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
+
+	r.IgnoreErrors(apierrors.IsNotFound)
+
+	infos, err := r.Infos()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(infos) == 0 {
+		return nil, fmt.Errorf("the %s/%s does't exist in Karmada", args[0], args[1])
+	}
```

Comment:
```suggestion

	infos, err := r.Infos()
	if err != nil {
		return nil, err
	}
```

Reply:
If `r.Infos()` returns an empty slice, it will return an `IsNotFound` error.
```bash
$ karmadactl rebalance pods nginx
error: failed to get resource in Karmada: pods "nginx" not found
```